### PR TITLE
gui: Fix LLE module list

### PR DESCRIPTION
**Pull request by VelocityRa**
Also run `clang-format` on project.

### CODE REVIEW

File: vita3k/gui/src/settings_dialog.cpp [hunk 1]
```diff
@@ -195,12 +200,12 @@ void draw_settings_dialog(GuiState &gui, HostState &host) {
                 host.cfg.pref_path += '/';
             if (host.cfg.pref_path != host.pref_path) {
                 //if (change_pref_location(host.cfg.pref_path, host.pref_path)) { TODO
-                    host.pref_path = host.cfg.pref_path;
+                host.pref_path = host.cfg.pref_path;
```
Comment (VelocityRa) [inline vita3k/gui/src/settings_dialog.cpp]: These changes are in the wrong commit, but oh well.

File: vita3k/gui/src/imgui_impl_sdl_vulkan.cpp [hunk 2]
```diff
@@ -416,10 +416,22 @@ IMGUI_API void ImGui_ImplSdlVulkan_RenderDrawData(ImGui_VulkanState &state) {
...
+        0,
+        0,
+        0,
+        0,
+        1,
+        0,
+        -1,
+        -1,
+        0,
+        1,
```
Comment (1whatleytay) [inline vita3k/gui/src/imgui_impl_sdl_vulkan.cpp]: :eyes:

Comment (VelocityRa) [reply vita3k/gui/src/imgui_impl_sdl_vulkan.cpp]: 👀

File: vita3k/gui/src/settings_dialog.cpp [hunk 3]
```diff
@@ -94,18 +97,20 @@ void draw_settings_dialog(GuiState &gui, HostState &host) {
...
-                        continue;
-                    if (ImGui::Selectable(gui.modules_list[m].c_str(), module_select[m]) && modules == host.cfg.lle_modules.end())
-                        host.cfg.lle_modules.push_back(gui.modules_list[m]);
-                    else if (!module_select[m] && modules != host.cfg.lle_modules.end())
-                        host.cfg.lle_modules.erase(modules);
+                    const auto module = std::find(host.cfg.lle_modules.begin(), host.cfg.lle_modules.end(), gui.modules_list[m]);
+                    const bool module_existed = (module != host.cfg.lle_modules.end());
+                    if (module != host.cfg.lle_modules.end() && !gui.lle_modules_selected[m])
+                        if (!gui.module_search_bar.PassFilter(gui.modules_list[m].c_str()))
+                            continue;
```
Comment (Zangetsu38) [inline vita3k/gui/src/settings_dialog.cpp]: remove first line, that broken seach
```suggestion
                    if (!gui.module_search_bar.PassFilter(gui.modules_list[m].c_str()))
                        continue;
```

**Comment by Zangetsu38** [comment]
good job, works fine now :)

File: vita3k/gui/include/gui/state.h [hunk 4]
```diff
@@ -113,6 +114,7 @@ struct GuiState {
     std::string online_id;
 
     std::vector<std::string> modules_list;
+    std::vector<bool> modules_selected;
```
Comment (Zangetsu38) [inline vita3k/gui/include/gui/state.h]: put it just before modules_list ^^

File: vita3k/gui/src/settings_dialog.cpp [hunk 5]
```diff
@@ -77,6 +76,10 @@ void get_modules_list(GuiState &gui, HostState &host)
                 gui.modules_list.push_back(lle.filename().replace_extension().string());
         }
     }
+
+    gui.modules_selected.resize(gui.modules_list.size());
+    for (size_t i = 0; i < gui.modules_list.size(); ++i)
+        gui.modules_selected[i] = std::find(host.cfg.lle_modules.begin(), host.cfg.lle_modules.end(), gui.modules_list[i]) != host.cfg.lle_modules.end();
```
Comment (Zangetsu38) [inline vita3k/gui/src/settings_dialog.cpp]: secondary compared my message, for maybe run iut only if module folder is not empty/no exist ^^
```suggestion
        }
            gui.modules_selected.resize(gui.modules_list.size());
    for (size_t i = 0; i < gui.modules_list.size(); ++i)
        gui.modules_selected[i] = std::find(host.cfg.lle_modules.begin(), host.cfg.lle_modules.end(), gui.modules_list[i]) != host.cfg.lle_modules.end();
    }
```

**Comment by VelocityRa** [comment]
Master is broken currently without this, so merging without further reviews.

**Pull request merged**